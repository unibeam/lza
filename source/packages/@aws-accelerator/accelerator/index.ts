/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

export * from './lib/accelerator';
export * from './lib/accelerator-stage';
export * from './lib/config-repository';
export * from './lib/pipeline';
export * from './lib/stacks/accelerator-stack';
export * from './lib/stacks/accounts-stack';
export * from './lib/stacks/bootstrap-stack';
export * from './lib/stacks/diagnostics-pack-stack';
export * from './lib/stacks/network-stacks/network-associations-stack/network-associations-stack';
export * from './lib/stacks/network-stacks/network-prep-stack/network-prep-stack';
export * from './lib/stacks/network-stacks/network-vpc-stack/network-vpc-stack';
export * from './lib/stacks/operations-stack';
export * from './lib/stacks/organizations-stack';
export * from './lib/stacks/pipeline-stack';
export * from './lib/stacks/prepare-stack';
export * from './lib/stacks/finalize-stack';
export * from './lib/stacks/security-audit-stack';
export * from './lib/stacks/security-stack';
export * from './lib/toolkit';
export * from './lib/validate-environment-config';
