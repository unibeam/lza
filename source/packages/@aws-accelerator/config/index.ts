/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

export * from './lib/accounts-config';
export * from './lib/common';
export * from './lib/customizations-config';
export * from './lib/global-config';
export * from './lib/iam-config';
export * from './lib/network-config';
export * from './lib/organization-config';
export * from './lib/replacements-config';
export * from './lib/security-config';
export * from './lib/models/accounts-config';
export * from './lib/models/global-config';
export * from './lib/models/iam-config';
export * from './lib/models/organization-config';
export * from './lib/models/security-config';
export * from './validator/accounts-config-validator';
export * from './validator/customizations-config-validator';
export * from './validator/global-config-validator';
export * from './validator/iam-config-validator';
export * from './validator/organization-config-validator';
export * from './validator/network-config-validator/network-config-validator';
export * from './validator/security-config-validator';
export * from './validator/replacements-config-validator';
export * from './validator/common/common-validator-functions';
