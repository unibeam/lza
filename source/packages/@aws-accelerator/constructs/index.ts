/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

export * from './lib/aws-accelerator/get-accelerator-metadata';
export * from './lib/aws-budgets/budget-definition';
export * from './lib/aws-cloudformation/get-resource-type';
export * from './lib/aws-cloudwatch-logs/cloudwatch-log-data-protection';
export * from './lib/aws-cloudwatch-logs/cloudwatch-destination';
export * from './lib/aws-cloudwatch-logs/cloudwatch-log-group';
export * from './lib/aws-cloudwatch-logs/cloudwatch-logs-subscription-filter';
export * from './lib/aws-configservice/config-tags';
export * from './lib/aws-configservice/config-aggregation';
export * from './lib/aws-configservice/config-recorder';
export * from './lib/data-perimeter/detect-resource-policy';
export * from './lib/data-perimeter/remediate-resource-policy';
export * from './lib/data-perimeter/remediation-ssm-document';
export * from './lib/aws-controltower/create-accounts';
export * from './lib/aws-controltower/create-enabled-controls';
export * from './lib/aws-cur/report-definition';
export * from './lib/aws-detective/detective-graph-config';
export * from './lib/aws-detective/detective-members';
export * from './lib/aws-detective/detective-organization-admin-account';
export * from './lib/aws-directconnect/direct-connect-gateway';
export * from './lib/aws-directconnect/gateway-association';
export * from './lib/aws-directconnect/virtual-interface';
export * from './lib/aws-directory-service/active-directory';
export * from './lib/aws-directory-service/active-directory-configuration';
export * from './lib/aws-directory-service/active-directory-log-subscription';
export * from './lib/aws-directory-service/active-directory-resolver-rule';
export * from './lib/aws-directory-service/share-active-directory';
export * from './lib/aws-ec2/account-warming';
export * from './lib/aws-ec2/cross-account-route';
export * from './lib/aws-ec2/customer-gateway';
export * from './lib/aws-ec2/delete-default-vpc';
export * from './lib/aws-ec2/dhcp-options';
export * from './lib/aws-ec2/ebs-encryption';
export * from './lib/aws-ec2/firewall';
export * from './lib/aws-ec2/firewall-config-replacements';
export * from './lib/aws-ec2/firewall-instance';
export * from './lib/aws-ec2/firewall-asg';
export * from './lib/aws-ec2/ipam';
export * from './lib/aws-ec2/ipam-pool';
export * from './lib/aws-ec2/ipam-organization-admin-account';
export * from './lib/aws-ec2/ipam-scope';
export * from './lib/aws-ec2/ipam-subnet';
export * from './lib/aws-ec2/prefix-list';
export * from './lib/aws-ec2/prefix-list-route';
export * from './lib/aws-ec2/route-table';
export * from './lib/aws-ec2/subnet-id-lookup';
export * from './lib/aws-ec2/transit-gateway';
export * from './lib/aws-ec2/transit-gateway-connect';
export * from './lib/aws-ec2/transit-gateway-peering';
export * from './lib/aws-ec2/transit-gateway-prefix-list-reference';
export * from './lib/aws-ec2/transit-gateway-route-table';
export * from './lib/aws-ec2/transit-gateway-static-route';
export * from './lib/aws-ec2/vpc';
export * from './lib/aws-ec2/vpc-endpoint';
export * from './lib/aws-ec2/vpc-id-lookup';
export * from './lib/aws-ec2/vpc-peering';
export * from './lib/aws-ec2/vpn-connection';
export * from './lib/aws-events/move-account-rule';
export * from './lib/aws-events/new-cloudwatch-log-event-rule';
export * from './lib/aws-events/revert-scp-changes';
export * from './lib/aws-elasticloadbalancingv2/gateway-load-balancer';
export * from './lib/aws-elasticloadbalancingv2/nlb-addresses';
export * from './lib/aws-guardduty/guardduty-detector-config';
export * from './lib/aws-guardduty/guardduty-members';
export * from './lib/aws-guardduty/guardduty-organization-admin-account';
export * from './lib/aws-guardduty/guardduty-publishing-destination';
export * from './lib/aws-auditmanager/auditmanager-organization-admin-account';
export * from './lib/aws-auditmanager/auditmanager-reports-destination';
export * from './lib/aws-detective/detective-members';
export * from './lib/aws-detective/detective-organization-admin-account';
export * from './lib/aws-detective/detective-graph-config';
export * from './lib/aws-iam/password-policy';
export * from './lib/aws-iam/service-linked-role';
export * from './lib/aws-identity-center/identity-center-instance';
export * from './lib/aws-identity-center/identity-center-organization-admin-account';
export * from './lib/aws-identity-center/identity-center-get-permission-set-role-arn';
export * from './lib/aws-identity-center/identity-center-assignments';
export * from './lib/aws-firehose/cloudwatch-to-s3-firehose';
export * from './lib/aws-fms/fms-notification-channel';
export * from './lib/aws-fms/fms-organization-admin-account';
export * from './lib/aws-kms/key-encryption';
export * from './lib/aws-kms/key-lookup';
export * from './lib/aws-macie/macie-export-config-classification';
export * from './lib/aws-macie/macie-members';
export * from './lib/aws-macie/macie-organization-admin-account';
export * from './lib/aws-macie/macie-session';
export * from './lib/aws-networkfirewall/firewall';
export * from './lib/aws-networkfirewall/policy';
export * from './lib/aws-networkfirewall/rule-group';
export * from './lib/aws-opt-in-regions/enable-opt-in-regions';
export * from './lib/aws-organizations/account';
export * from './lib/aws-organizations/create-accounts';
export * from './lib/aws-organizations/enable-aws-service-access';
export * from './lib/aws-organizations/enable-policy-type';
export * from './lib/aws-organizations/move-accounts';
export * from './lib/aws-organizations/organizational-units';
export * from './lib/aws-organizations/policy';
export * from './lib/aws-organizations/policy-attachment';
export * from './lib/aws-organizations/register-delegated-administrator';
export * from './lib/aws-organizations/validate-scp-count';
export * from './lib/aws-ram/enable-sharing-with-aws-organization';
export * from './lib/aws-ram/resource-share';
export * from './lib/aws-ram/share-subnet-tags';
export * from './lib/aws-route-53-resolver/firewall-domain-list';
export * from './lib/aws-route-53-resolver/firewall-rule-group';
export * from './lib/aws-route-53-resolver/query-logging-config';
export * from './lib/aws-route-53-resolver/resolver-endpoint';
export * from './lib/aws-route-53-resolver/resolver-rule';
export * from './lib/aws-route-53/associate-hosted-zones';
export * from './lib/aws-route-53/hosted-zone';
export * from './lib/aws-route-53/record-set';
export * from './lib/aws-s3/bucket';
export * from './lib/aws-s3/bucket-encryption';
export * from './lib/aws-s3/bucket-replication';
export * from './lib/aws-s3/bucket-policy';
export * from './lib/aws-s3/bucket-prefix';
export * from './lib/aws-s3/central-logs-bucket';
export * from './lib/aws-s3/public-access-block';
export * from './lib/aws-s3/validate-bucket';
export * from './lib/aws-securityhub/securityhub-members';
export * from './lib/aws-securityhub/securityhub-organization-admin-account';
export * from './lib/aws-securityhub/securityhub-standards';
export * from './lib/aws-securityhub/securityhub-region-aggregation';
export * from './lib/aws-servicecatalog/get-portfolio-id';
export * from './lib/aws-service-quota/limits-service-quota-definition';
export * from './lib/aws-servicecatalog/share-portfolio-with-org';
export * from './lib/aws-servicecatalog/propagate-portfolio-associations';
export * from './lib/aws-ssm/document';
export * from './lib/aws-ssm/inventory';
export * from './lib/aws-ssm/session-manager-settings';
export * from './lib/aws-ssm/policy-attachment';
export * from './lib/aws-ssm/put-ssm-parameter';
export * from './lib/aws-ssm/ssm-parameter-lookup';
export * from './lib/aws-events/security-hub-events-log';
export * from './lib/aws-ec2/create-launch-template';
export * from './lib/aws-autoscaling/create-autoscaling-group';
export * from './lib/aws-elasticloadbalancingv2/target-group';
export * from './lib/aws-elasticloadbalancingv2/network-load-balancer';
export * from './lib/aws-elasticloadbalancingv2/application-load-balancer';
export * from './lib/aws-certificate-manager/certificate';
export * from './lib/lza-custom-resource';
export * from './lib/lza-lambda';
