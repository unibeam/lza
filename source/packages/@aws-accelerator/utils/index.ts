/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

export * from './lib/common-resources';
export * from './lib/common-functions';
export * from './lib/logger';
export * from './lib/policy-replacements';
export * from './lib/set-organizations-client';
export * from './lib/ssm-parameter-path';
export * from './lib/throttle';
export * from './lib/load-organization-config';
export * from './lib/get-template';
export * from './lib/diff-stack';
export * from './lib/regions';
export * from './lib/set-token-preferences';
export * from './lib/evaluate-limits';
export * from './lib/common-types';
export * from './lib/control-tower';

//
// Common integration test utilities
//
export * from './lib/test-util/common/resources';
export * from './lib/test-util/common/integration-test';

//
// Security integration test suite
//
export * from './lib/test-util/common/test-suite';
