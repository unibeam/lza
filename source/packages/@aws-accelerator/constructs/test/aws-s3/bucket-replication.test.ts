/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

import * as cdk from 'aws-cdk-lib';
import { BucketReplication } from '@aws-accelerator/constructs';
import { snapShotTest } from '../snapshot-test';
import { describe, expect, test } from '@jest/globals';

const testNamePrefix = 'Construct(BucketReplication): ';

//Initialize stack for snapshot test and resource configuration test
const stack = new cdk.Stack();

new BucketReplication(stack, 'BucketReplication', {
  source: {
    bucketName: `aws-accelerator-macie-${cdk.Aws.ACCOUNT_ID}-${cdk.Aws.REGION}`,
  },
  destination: {
    bucketName: `aws-accelerator-central-logs-bucket`,
    accountId: cdk.Aws.ACCOUNT_ID,
    keyArn: `arn:aws:kms:us-east-1:${cdk.Aws.ACCOUNT_ID}:key/ksm-key-arn`,
  },
  kmsKey: new cdk.aws_kms.Key(stack, 'CWLKey', {}),
  logRetentionInDays: 3653,
  useExistingRoles: false,
  acceleratorPrefix: 'AWSAccelerator',
});

new BucketReplication(stack, 'BucketReplicationExistingIam', {
  source: {
    bucket: new cdk.aws_s3.Bucket(stack, 'SourceBucket', {
      encryption: cdk.aws_s3.BucketEncryption.KMS,
      encryptionKey: new cdk.aws_kms.Key(stack, 'SourceBucketExistingIam'),
    }),
  },
  destination: {
    bucketName: `aws-accelerator-central-logs-bucket-existing-iam`,
    accountId: cdk.Aws.ACCOUNT_ID,
    keyArn: `arn:aws:kms:us-east-1:${cdk.Aws.ACCOUNT_ID}:key/ksm-key-arn`,
  },
  kmsKey: new cdk.aws_kms.Key(stack, 'CWLKeyExistingIam', {}),
  logRetentionInDays: 3653,
  useExistingRoles: true,
  acceleratorPrefix: 'AWSAccelerator',
});
/**
 * BucketReplication construct test
 */
describe('BucketReplication', () => {
  snapShotTest(testNamePrefix, stack);
});

test('should throw an exception when source bucket name and bucket are present', () => {
  function s3SourceBucketError1() {
    new BucketReplication(stack, 'BucketReplicationSourceBucketError1', {
      source: {
        bucket: new cdk.aws_s3.Bucket(stack, 'SourceBucketError1', {}),
        bucketName: `aws-accelerator-macie-${cdk.Aws.ACCOUNT_ID}-${cdk.Aws.REGION}`,
      },
      destination: {
        bucketName: `aws-accelerator-central-logs-bucket`,
        accountId: cdk.Aws.ACCOUNT_ID,
        keyArn: `arn:aws:kms:us-east-1:${cdk.Aws.ACCOUNT_ID}:key/ksm-key-arn`,
      },
      kmsKey: new cdk.aws_kms.Key(stack, 'CWLKeySourceBucketError1', {}),
      logRetentionInDays: 3653,
      useExistingRoles: false,
      acceleratorPrefix: 'AWSAccelerator',
    });
  }

  const errMsg = 'Source bucket or source bucketName (only one property) should be defined.';
  expect(s3SourceBucketError1).toThrow(new Error(errMsg));
});

test('should throw an exception when source bucket name and bucket are not defined', () => {
  function s3SourceBucketError2() {
    new BucketReplication(stack, 'BucketReplicationSourceBucketError2', {
      source: {
        bucket: undefined,
        bucketName: undefined,
      },
      destination: {
        bucketName: `aws-accelerator-central-logs-bucket`,
        accountId: cdk.Aws.ACCOUNT_ID,
        keyArn: `arn:aws:kms:us-east-1:${cdk.Aws.ACCOUNT_ID}:key/ksm-key-arn`,
      },
      kmsKey: new cdk.aws_kms.Key(stack, 'CWLKeySourceBucketError2', {}),
      logRetentionInDays: 3653,
      useExistingRoles: false,
      acceleratorPrefix: 'AWSAccelerator',
    });
  }

  const errMsg = 'Source bucket or source bucketName property must be defined when using bucket replication.';
  expect(s3SourceBucketError2).toThrow(new Error(errMsg));
});
