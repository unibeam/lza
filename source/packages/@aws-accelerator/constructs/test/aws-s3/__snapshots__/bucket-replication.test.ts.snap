// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BucketReplication Construct(BucketReplication):  Snapshot Test 1`] = `
{
  "Resources": {
    "BucketReplicationAwsAcceleratorCentralLogsBucketReplicationRole184AC22C": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "s3.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Path": "/service-role/",
      },
      "Type": "AWS::IAM::Role",
    },
    "BucketReplicationAwsAcceleratorCentralLogsBucketReplicationRoleDefaultPolicy5AD1044F": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObjectLegalHold",
                "s3:GetObjectRetention",
                "s3:GetObjectVersion",
                "s3:GetObjectVersionAcl",
                "s3:GetObjectVersionForReplication",
                "s3:GetObjectVersionTagging",
                "s3:GetReplicationConfiguration",
                "s3:ListBucket",
                "s3:ReplicateDelete",
                "s3:ReplicateObject",
                "s3:ReplicateTags",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::aws-accelerator-macie-",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      "-",
                      {
                        "Ref": "AWS::Region",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::aws-accelerator-macie-",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      "-",
                      {
                        "Ref": "AWS::Region",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "s3:GetBucketVersioning",
                "s3:GetObjectVersionTagging",
                "s3:ObjectOwnerOverrideToBucketOwner",
                "s3:PutBucketVersioning",
                "s3:ReplicateDelete",
                "s3:ReplicateObject",
                "s3:ReplicateTags",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":s3:::aws-accelerator-central-logs-bucket/*",
                  ],
                ],
              },
            },
            {
              "Action": "kms:Encrypt",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:kms:us-east-1:",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":key/ksm-key-arn",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "BucketReplicationAwsAcceleratorCentralLogsBucketReplicationRoleDefaultPolicy5AD1044F",
        "Roles": [
          {
            "Ref": "BucketReplicationAwsAcceleratorCentralLogsBucketReplicationRole184AC22C",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "BucketReplicationDF143956": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "CustomS3PutBucketReplicationCustomResourceProviderLogGroup6A67905E",
      ],
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomS3PutBucketReplicationCustomResourceProviderHandler1D75398C",
            "Arn",
          ],
        },
        "destinationAccountId": {
          "Ref": "AWS::AccountId",
        },
        "destinationBucketArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":s3:::aws-accelerator-central-logs-bucket",
            ],
          ],
        },
        "destinationBucketKeyArn": {
          "Fn::Join": [
            "",
            [
              "arn:aws:kms:us-east-1:",
              {
                "Ref": "AWS::AccountId",
              },
              ":key/ksm-key-arn",
            ],
          ],
        },
        "prefix": "",
        "replicationRoleArn": {
          "Fn::GetAtt": [
            "BucketReplicationAwsAcceleratorCentralLogsBucketReplicationRole184AC22C",
            "Arn",
          ],
        },
        "sourceBucketName": {
          "Fn::Join": [
            "",
            [
              "aws-accelerator-macie-",
              {
                "Ref": "AWS::AccountId",
              },
              "-",
              {
                "Ref": "AWS::Region",
              },
            ],
          ],
        },
      },
      "Type": "Custom::S3PutBucketReplication",
      "UpdateReplacePolicy": "Delete",
    },
    "BucketReplicationExistingIamA4FEEDD2": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "CustomS3PutBucketReplicationCustomResourceProviderLogGroup6A67905E",
      ],
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomS3PutBucketReplicationCustomResourceProviderHandler1D75398C",
            "Arn",
          ],
        },
        "destinationAccountId": {
          "Ref": "AWS::AccountId",
        },
        "destinationBucketArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":s3:::aws-accelerator-central-logs-bucket-existing-iam",
            ],
          ],
        },
        "destinationBucketKeyArn": {
          "Fn::Join": [
            "",
            [
              "arn:aws:kms:us-east-1:",
              {
                "Ref": "AWS::AccountId",
              },
              ":key/ksm-key-arn",
            ],
          ],
        },
        "prefix": "",
        "replicationRoleArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":iam::",
              {
                "Ref": "AWS::AccountId",
              },
              ":role/AWSAcceleratorS3ReplicationRole",
            ],
          ],
        },
        "sourceBucketName": {
          "Ref": "SourceBucketDDD2130A",
        },
      },
      "Type": "Custom::S3PutBucketReplication",
      "UpdateReplacePolicy": "Delete",
    },
    "CWLKeyB9201973": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "KeyPolicy": {
          "Statement": [
            {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
    "CWLKeyExistingIamDCB4F720": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "KeyPolicy": {
          "Statement": [
            {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
    "CustomS3PutBucketReplicationCustomResourceProviderHandler1D75398C": {
      "DependsOn": [
        "CustomS3PutBucketReplicationCustomResourceProviderRole1C378488",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "REPLACED-GENERATED-NAME.zip",
        },
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "CustomS3PutBucketReplicationCustomResourceProviderRole1C378488",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3PutBucketReplicationCustomResourceProviderLogGroup6A67905E": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "KmsKeyId": {
          "Fn::GetAtt": [
            "CWLKeyB9201973",
            "Arn",
          ],
        },
        "LogGroupName": {
          "Fn::Join": [
            "",
            [
              "/aws/lambda/",
              {
                "Ref": "CustomS3PutBucketReplicationCustomResourceProviderHandler1D75398C",
              },
            ],
          ],
        },
        "RetentionInDays": 3653,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "CustomS3PutBucketReplicationCustomResourceProviderRole1C378488": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
        "Policies": [
          {
            "PolicyDocument": {
              "Statement": [
                {
                  "Action": [
                    "iam:PassRole",
                    "s3:PutLifecycleConfiguration",
                    "s3:PutReplicationConfiguration",
                    "s3:PutBucketVersioning",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "S3PutReplicationConfigurationTaskActions",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "Inline",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "SourceBucketDDD2130A": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "KMSMasterKeyID": {
                  "Fn::GetAtt": [
                    "SourceBucketExistingIam4DBB8957",
                    "Arn",
                  ],
                },
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "SourceBucketExistingIam4DBB8957": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "KeyPolicy": {
          "Statement": [
            {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
